src/main/java/org/terasology/scenario/internal/systems/ActionEventSystem.java

```diff
@@ -158,4 +177,129 @@ public void onEventTriggerEvent(EventTriggerEvent event, EntityRef entity, Scena
+
+        EvaluateItemPrefabEvent itemEvaluateEvent = new EvaluateItemPrefabEvent(event.informationEntity);
+        variables.get("item").send(itemEvaluateEvent);
+        Prefab itemPrefab = itemEvaluateEvent.getResult();
+
+        EvaluateIntEvent intEvaluateEvent = new EvaluateIntEvent(event.informationEntity);
+        variables.get("amount").send(intEvaluateEvent);
+        int amount = intEvaluateEvent.getResult();
+
+        ScenarioValuePlayerComponent.PlayerType player = variables.get("player").getComponent(ScenarioValuePlayerComponent.class).type;
+
+        if (player == ScenarioValuePlayerComponent.PlayerType.TRIGGERING_PLAYER) {
+            EntityRef playerEnt = event.informationEntity.getComponent(InfoTriggeringEntityComponent.class).entity;
+
+            List<EntityRef> items = playerEnt.getComponent(InventoryComponent.class).itemSlots;
+
+            for (EntityRef e : items) {
+                if (e.exists() && e.getParentPrefab().exists() && e.getParentPrefab().equals(itemPrefab)) {
+                    e.destroy();
+                    amount--;
+                }
+
+                if (amount <= 0) {
+                    break;
+                }
+            }
+
+            playerEnt.saveComponent(playerEnt.getComponent(InventoryComponent.class));
+        }
+    }
+
+    @ReceiveEvent //Take Block
+    public void onEventTriggerEvent(EventTriggerEvent event, EntityRef entity, ScenarioSecondaryTakeBlockComponent action) {
+        Map<String, EntityRef> variables = entity.getComponent(ScenarioArgumentContainerComponent.class).arguments;
+
+        EvaluateBlockEvent evaluateBlockEvent = new EvaluateBlockEvent(event.informationEntity);
+        variables.get("block").send(evaluateBlockEvent);
+        BlockFamily blockFamily = evaluateBlockEvent.getResult();
+
+        EvaluateIntEvent intEvaluateEvent = new EvaluateIntEvent(event.informationEntity);
+        variables.get("amount").send(intEvaluateEvent);
+        int amount = intEvaluateEvent.getResult();
+
+        ScenarioValuePlayerComponent.PlayerType player = variables.get("player").getComponent(ScenarioValuePlayerComponent.class).type;
+
+        if (player == ScenarioValuePlayerComponent.PlayerType.TRIGGERING_PLAYER) {
+            EntityRef playerEnt = event.informationEntity.getComponent(InfoTriggeringEntityComponent.class).entity;
+
+            List<EntityRef> items = playerEnt.getComponent(InventoryComponent.class).itemSlots;
+
+            for (EntityRef e : items) {
+                if (e.hasComponent(BlockItemComponent.class)) {
+                    if (e.getComponent(BlockItemComponent.class).blockFamily.equals(blockFamily)) {
+                        ItemComponent itemComponent = e.getComponent(ItemComponent.class);
+                        if (itemComponent.stackCount > amount) {
+                            itemComponent.stackCount-= amount;

```

Comment:
Will check it out and fix up the inventory changes to utilize the inventory manager instead.